torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -1481,6 +1548,14 @@ def _reshard(
             free_unsharded_flat_params,
         ):
             handle.reshard(free_unsharded_flat_param)
```

Comment:
it still calls record_stream in the reshard(), should it be removed if self.limit_all_gathers?

Reply:
hmm, I guess it is fine..? because we slowed down CPU thread anyway, it is OK to let the record stream still hold the memory

Reply:
I think we need to still call `record_stream()` if `self.limit_all_gathers == True`. Otherwise, the memory could still be freed and reused too early (e.g. before we even get a chance to synchronize the CPU thread in `_unshard()`).